### ISSUE TITLE
Use a tag or a commit hash when setting up dependencies

### ISSUE THREAD

**Issue by hellais** [TheTorProject/ooni-sysadmin#95]
Currently in many places throughout this repository the deployment script looks something like:

```
git clone https://github.com/org/some-component
install-some-component.sh
```

This is bad because it means that if `some-component` makes breaking changes to the deployment we will end up with broken deployment scripts.

We should ensure that every time we install a package and every time we clone a git repository a tag is specified or a commit hash.

This way we can ensure that the deployment script will always work independently from upstream changes. When we need to update to a later version we just have to change the deployment script to use the new commit hash or the new tag.

**Comment by anadahz** [comment]
Indeed that will be very useful.
Currently most ansible playbook are being deployed with the latest master (or the latest version)  in mind which is usually what we are doing since the master branch in these releases has been tested extensively.

Shall we use a specific tag or do we go for the last know tested and deployed commit hash?

A more advanced way is (CI like) test build (with Travis?) on the latest version (since we are highly interest in using the software's latest version, especially in the examples of the comment https://github.com/TheTorProject/ooni-sysadmin/issues/95#issue-214376547 ) of software and be alarmed of any bugs or potentially broken deployments.

**Comment by hellais** [comment]
> Shall we use a specific tag or do we go for the last know tested and deployed commit hash?

You should use whatever commitsh or tag you are making the deployment script for. If you are targeting the current master instead of just plain cloning you clone with as target the commitsh of the current head.
If you are targeting a specific release of the software, then you target that release.

> A more advanced way is (CI like) test build (with Travis?) on the latest version (since we are highly interest in using the software's latest version, especially in the examples of #95 (comment) ) of software and be alarmed of any bugs or potentially broken deployments.

CI is and testing of the current master is a separate problem from that of deployment. Most software project part of the OONI ecosystem already have unittesting and some even continuous delivery to staging (such as ooni-measurements and ooni-probe), but I don't think should overlap with the problem of deployment.

A deployment is done on something that is tested and that in the future you should be able to deterministically re-deploy the same exactly piece of code by checking out the deployment script at the correct commit.

**Comment by anadahz** [comment]
@hellais I agree with https://github.com/TheTorProject/ooni-sysadmin/issues/95#issuecomment-287336747.
I guess this ticket should be renamed to: Change all dependencies to use a specific version, git tar or commit hash.

**Comment by darkk** [comment]
I'm unsure if this ticket is actionable as-is. It sound like "do good thing and don't do bad things" ticket, so maybe it should be closed. However, there are some deployment scripts where version [is not set](https://github.com/ooni/sysadmin/blob/master/ansible/roles/postgresql/tasks/install.yml), so it's not 100% clear to me.
